iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -29,58 +27,23 @@ namespace popo
 template <typename T, typename H>
 class PublisherInterface;
 
-namespace internal
-{
-/// @brief helper struct for sample
-template <typename T, typename H>
-struct SamplePrivateData
-{
-    SamplePrivateData(cxx::unique_ptr<T>&& sampleUniquePtr, PublisherInterface<T, H>& publisher) noexcept;
-
-    SamplePrivateData(SamplePrivateData&& rhs) noexcept = default;
-    SamplePrivateData& operator=(SamplePrivateData&& rhs) noexcept = default;
-
-    SamplePrivateData(const SamplePrivateData&) = delete;
-    SamplePrivateData& operator=(const SamplePrivateData&) = delete;
-
-    cxx::unique_ptr<T> sampleUniquePtr;
-    std::reference_wrapper<PublisherInterface<T, H>> publisherRef;
-};
-
-/// @brief specialization of helper struct for sample for const T
-template <typename T, typename H>
-struct SamplePrivateData<const T, H>
-{
-    SamplePrivateData(cxx::unique_ptr<const T>&& sampleUniquePtr) noexcept;
-
-    SamplePrivateData(SamplePrivateData&& rhs) noexcept = default;
-    SamplePrivateData& operator=(SamplePrivateData&& rhs) noexcept = default;
-
-    SamplePrivateData(const SamplePrivateData&) = delete;
-    SamplePrivateData& operator=(const SamplePrivateData&) = delete;
-
-    cxx::unique_ptr<const T> sampleUniquePtr;
-};
-} // namespace internal
-
 ///
 /// @brief The Sample class is a mutable abstraction over types which are written to loaned shared memory.
 /// These samples are publishable to the iceoryx system.
 ///
 template <typename T, typename H = cxx::add_const_conditionally_t<mepoo::NoUserHeader, T>>
-class Sample
+class Sample : public SmartChunk<PublisherInterface<T, H>, T, H>
 {
     static_assert(std::is_const<T>::value == std::is_const<H>::value,
                   "The type `T` and the user-header `H` must be equal in their const qualifier to ensure the same "
                   "access restrictions for the user-header as for the sample data!");
 
-    /// @brief Helper type to enable the constructor for the publisher, i.e. when T has no const qualifier
     template <typename S, typename TT>
-    using ForPublisherOnly = std::enable_if_t<std::is_same<S, TT>::value && !std::is_const<TT>::value, S>;
+    using ForProducerOnly = std::enable_if_t<std::is_same<S, TT>::value && !std::is_const<TT>::value, S>;
```

Comment:
I did not test it, bud doesn't this also work?
```cpp
using SmartChunk<PublisherInterface<T, H>, T, H>::ForProducerOnly;
```

Reply:
No it does not. Dont know the exact reason though.

Reply:
I think the following should work
```cpp
using Base_t = SmartChunk<PublisherInterface<T, H>, T, H>;

template <typename S, typename TT>
using ForProducerOnly = typename Base_t::template ForProducerOnly<S, TT>;
```